src/kademlia/dht_state.cpp

```diff
@@ -81,6 +90,13 @@ namespace
 			ret.nodes6 = detail::read_endpoint_list<udp::endpoint>(nodes);
 #endif
 
+		if (bdecode_node const st = e.dict_find_string("storage-state"))
+		{
+			char const* ptr = st.string_ptr();
```

Comment:
this is meant to be a "blob" defined by the storage implementation? would there need to be protection from passing in the blob from another storage class?